db/structure.sql

```diff
@@ -2650,6 +2513,196 @@ COMMENT ON COLUMN public.reporting_patient_states.htn_treatment_outcome_in_quart
 COMMENT ON COLUMN public.reporting_patient_states.titrated IS 'True, if the patient had an increase in dosage of any hypertension drug in a visit this month.';
 
 
+--
+-- Name: reporting_facility_state_groups; Type: MATERIALIZED VIEW; Schema: public; Owner: -
+--
+
+CREATE MATERIALIZED VIEW public.reporting_facility_state_groups AS
+ WITH registered_patients AS (
+         SELECT reporting_patient_states.registration_facility_region_id AS facility_region_id,
+            reporting_patient_states.month_date,
+            count(*) AS monthly_registrations_all,
+            count(*) FILTER (WHERE (reporting_patient_states.hypertension = 'yes'::text)) AS monthly_registrations_htn_all,
+            count(*) FILTER (WHERE ((reporting_patient_states.hypertension = 'yes'::text) AND ((reporting_patient_states.gender)::text = 'female'::text))) AS monthly_registrations_htn_female,
+            count(*) FILTER (WHERE ((reporting_patient_states.hypertension = 'yes'::text) AND ((reporting_patient_states.gender)::text = 'male'::text))) AS monthly_registrations_htn_male,
+            count(*) FILTER (WHERE ((reporting_patient_states.hypertension = 'yes'::text) AND ((reporting_patient_states.gender)::text = 'transgender'::text))) AS monthly_registrations_htn_transgender,
+            count(*) FILTER (WHERE (reporting_patient_states.diabetes = 'yes'::text)) AS monthly_registrations_dm_all,
+            count(*) FILTER (WHERE ((reporting_patient_states.diabetes = 'yes'::text) AND ((reporting_patient_states.gender)::text = 'female'::text))) AS monthly_registrations_dm_female,
+            count(*) FILTER (WHERE ((reporting_patient_states.diabetes = 'yes'::text) AND ((reporting_patient_states.gender)::text = 'male'::text))) AS monthly_registrations_dm_male,
+            count(*) FILTER (WHERE ((reporting_patient_states.diabetes = 'yes'::text) AND ((reporting_patient_states.gender)::text = 'transgender'::text))) AS monthly_registrations_dm_transgender
+           FROM public.reporting_patient_states
+          WHERE (reporting_patient_states.months_since_registration = (0)::double precision)
+          GROUP BY reporting_patient_states.registration_facility_region_id, reporting_patient_states.month_date
+        )
+ SELECT registered_patients.facility_region_id,
+    registered_patients.month_date,
+    registered_patients.monthly_registrations_all,
+    registered_patients.monthly_registrations_htn_all,
+    registered_patients.monthly_registrations_htn_female,
+    registered_patients.monthly_registrations_htn_male,
+    registered_patients.monthly_registrations_htn_transgender,
+    registered_patients.monthly_registrations_dm_all,
+    registered_patients.monthly_registrations_dm_female,
+    registered_patients.monthly_registrations_dm_male,
+    registered_patients.monthly_registrations_dm_transgender
+   FROM registered_patients
+  ORDER BY registered_patients.month_date DESC
+  WITH NO DATA;
```

Comment:
This gives us grouped registrations by diagnosis. Need to see how long this matview takes to generate in prod to see if this particular view (and ancestor views) can be refreshed more frequently.